### PR TITLE
Adding snap instruction for add rack controller instruction.

### PR DESCRIPTION
**Pull request by xtrusia**
## Done

- Adding snap instruction for add rack controller instruction.
- Adding version confirmation for add rack controller instruction.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Steps for QA.

## Fixes

Fixes: # .

## Launchpad issue

Related Launchpad maas issue in the form `lp#1864097`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

**Comment by squidsoup** [comment]
@amylily1011 would you mind having a look here? Currently this looks like:

image

We could do what @xtrusia proposes here, or maybe there's a clearer way to differentiate the instructions for snaps vs deb installs?

**Comment by amylily1011** [comment]
Where can I find this message in the UI? @squidsoup

**Comment by squidsoup** [comment]
> Where can I find this message in the UI? @squidsoup

On the controllers view

**Comment by amylily1011** [comment]
@billwear is reviewing the content right now and he found some bug. We will let this sit for now until we resolve the bug.
https://bugs.launchpad.net/maas/+bug/1887812

File: legacy/src/app/partials/nodes-list.html [hunk 1]
```diff
@@ -716,9 +716,13 @@ <h4 class="page-header__dropdown-title">Add rack controller</h4>
...
 # Install the maas-rack-controller package.
 sudo apt install maas-rack-controller
+# Or if you use snap
+sudo snap install maas
 # Register the rack controller with this MAAS. If the rack controller (and machines)
 # don't have access to the URL, use a different IP address to allow connection.
-sudo maas-rack register --url {$ tabs.controllers.registerUrl $} --secret {$ tabs.controllers.registerSecret $}</code></pre>
+sudo maas-rack register --url {$ tabs.controllers.registerUrl $} --secret {$ tabs.controllers.registerSecret $}
+# Or if you use snap
+sudo /snap/maas/current/bin/maas-rack register --url {$ tabs.controllers.registerUrl $} --secret {$ tabs.controllers.registerSecret $}</code></pre>
```
Comment (sombrafam) [inline legacy/src/app/partials/nodes-list.html:726]: Why this maas-rack command is not in the system path like the mass command? The snap should be fixed to add it to the path.

File: legacy/src/app/partials/nodes-list.html [hunk 2]
```diff
@@ -715,6 +715,7 @@ <h4 class="page-header__dropdown-title">Add device</h4>
             <h4 class="page-header__dropdown-title">Add rack controller</h4>
             <pre><code># To add a new rack controller, SSH into the rack controller.
 # Install the maas-rack-controller package.
+# Please confirm maas version is the same as main rack controller first.
```
Comment (sombrafam) [inline legacy/src/app/partials/nodes-list.html:718]: Just saying: "Confirm maas version is..." should be enough, no need for 'Please'

File: legacy/src/app/partials/nodes-list.html (hunk 1 above)
Comment (albertodonato) [reply legacy/src/app/partials/nodes-list.html:726]: This is not the expected way to register a rackcontroller in the snap. The `maas` CLI should be used as:

```
sudo maas init rack --maas-url $url --secret $secret
```

For reference, this is the CLI help:

```
$ maas init rack --help
usage: maas init rack [-h] [--maas-url MAAS_URL] [--secret SECRET] [--force]

Initialise MAAS to run only a rack controller.

optional arguments:
  -h, --help           show this help message and exit
  --maas-url MAAS_URL  URL that MAAS should use for communicate from the nodes
                       to MAAS and other controllers of MAAS.
  --secret SECRET      Secret token required for the rack controller to talk
                       to the region controller(s). Only used when in 'rack'
                       mode.
  --force              Skip confirmation questions when initialization has
                       already been performed.
```

File: legacy/src/app/partials/nodes-list.html [hunk 3]
```diff
@@ -715,10 +715,15 @@ <h4 class="page-header__dropdown-title">Add device</h4>
...
```
Comment (amylily1011) [inline legacy/src/app/partials/nodes-list.html:718]: Please change the text in line 718 to:
'Confirm that the maas version is the the same is the main rack controller.'

**Comment by squidsoup** [comment]
@xtrusia I've updated the copy based on @albertodonato's suggestion, and will merge this now. Thanks for the contribution!

**Pull request merged**